json/src/main/java/io/airlift/json/SmileCodecFactory.java

```diff
@@ -0,0 +1,155 @@
+    public SmileCodecFactory(@Named("SMILE") Provider<ObjectMapper> smileObjectMapperProvider)
+    {
+        this(smileObjectMapperProvider, false);
+    }
+
+    public SmileCodecFactory(Provider<ObjectMapper> smileObjectMapperProvider, boolean prettyPrint)
+    {
+        this.objectMapperProvider = requireNonNull(smileObjectMapperProvider, "objectMapperProvider is null");
+        this.prettyPrint = prettyPrint;
+    }
+
+    public SmileCodecFactory prettyPrint()
+    {
+        return new SmileCodecFactory(objectMapperProvider, true);
+    }
+
+    public <T> JsonCodec<T> smileCodec(Class<T> type)
+    {
+        requireNonNull(type, "type is null");
+
+        return new JsonCodec<>(createObjectMapper(), type);
+    }
+
+    public <T> JsonCodec<T> smileCodec(Type type)
+    {
+        requireNonNull(type, "type is null");
+
+        return new JsonCodec<>(createObjectMapper(), type);
+    }
+
+    public <T> JsonCodec<T> smileCodec(TypeToken<T> type)
+    {
+        requireNonNull(type, "type is null");
+
+        return new JsonCodec<>(createObjectMapper(), type.getType());
+    }
+
+    public <T> JsonCodec<List<T>> listSmileCodec(Class<T> type)
+    {
+        requireNonNull(type, "type is null");
+
+        Type listType = new TypeToken<List<T>>() {}
+                .where(new TypeParameter<T>() {}, type)
+                .getType();
+
+        return new JsonCodec<>(createObjectMapper(), listType);
+    }
+
+    public <T> JsonCodec<List<T>> listSmileCodec(JsonCodec<T> type)
+    {
+        requireNonNull(type, "type is null");
+
+        Type listType = new TypeToken<List<T>>() {}
+                .where(new TypeParameter<T>() {}, type.getTypeToken())
+                .getType();
+
+        return new JsonCodec<>(createObjectMapper(), listType);
+    }
+
+    public <K, V> JsonCodec<Map<K, V>> mapSmileCodec(Class<K> keyType, Class<V> valueType)
+    {
+        requireNonNull(keyType, "keyType is null");
+        requireNonNull(valueType, "valueType is null");
+
+        Type mapType = new TypeToken<Map<K, V>>() {}
+                .where(new TypeParameter<K>() {}, keyType)
+                .where(new TypeParameter<V>() {}, valueType)
+                .getType();
+
+        return new JsonCodec<>(createObjectMapper(), mapType);
+    }
+
+    public <K, V> JsonCodec<Map<K, V>> mapSmileCodec(Class<K> keyType, JsonCodec<V> valueType)
+    {
+        requireNonNull(keyType, "keyType is null");
+        requireNonNull(valueType, "valueType is null");
+
+        Type mapType = new TypeToken<Map<K, V>>() {}
+                .where(new TypeParameter<K>() {}, keyType)
+                .where(new TypeParameter<V>() {}, valueType.getTypeToken())
+                .getType();
+
+        return new JsonCodec<>(createObjectMapper(), mapType);
+    }
+
+    private ObjectMapper createObjectMapper()
+    {
+        ObjectMapper objectMapper = null;
+
+        // Work around for http://code.google.com/p/google-guice/issues/detail?id=627
```

Comment:
It looks like we don't need this anymore. Can you remove it from the other one in a commit before this one?